### ISSUE TITLE
Inquiry about Model Weights

### ISSUE THREAD

**Issue by ahmadsuhail1** [benkeel/VAE_lung_lesion_BMVC#2]
Hey Guys, are there any model weights available, so i can test this out quickly.

**Comment by benkeel** [comment]
Hi! Thanks for the interest in the code, unfortunately in hindsight I didn't make it that easy to use

I've uploaded the image data (as numpy) and some parameters here (expiring in 30 days):
https://leeds365-my.sharepoint.com/:f:/g/personal/mm17b2k_leeds_ac_uk/EtDf7zxgKRhGjHp_z5B_BRUBEc-Nbfv13AflgKyW2pJa-g?e=l0Jc7I
password = password

Also added the labels and an excel spreadsheet of my experiments/results, which shows good candidates for the VAE and MLP model hyperparameters

On "Clustering and MLP/Clustering_initial" it has a simple way to load the model

If you do end up writing a script to test the model I'd be happy to add it to the repo 

Hope this helps,
Ben

**Comment by zh1jaw** [comment]
Hi Ben,
Sorry to trouble you again, but I was wondering if it would be possible to also get access to the pretrained model for testing? I’d really appreciate it!
Thank you, 
Jaw

**Comment by benkeel** [comment]
Hi Jaw,

Thanks for reaching out!

I've shared the data, and gaussian VAE parameters for the best model before and after fine tuning using the MLP here:
https://leeds365-my.sharepoint.com/:f:/g/personal/mm17b2k_leeds_ac_uk/Egs98f75ttFPtMR9OSDQ1PUBpfV_3LRHukXtqqhVQdgBSA?e=LA19rq

I think this was the final run saved in VAE_joint_loss_mal_benign.py and so the hyperparameters are: 
parameter_space = {"HU_UpperBound":[400], 
                   "HU_LowerBound":[-500], 
                   "base":[128],       
                   "latent_size": [4], 
                   "annealing":[0],      
                   "ssim_indicator":[1],     
                   "batch_size":[64],   
                   "alpha":[0.5],  
                   "beta":[1.5], 
                   "lr":[1e-5],    
                   "ssim_scalar":[2],            
                   "recon_scale_factor":[1]}

With this I hope it should be easy enough to train some MLPs even on a CPU - see VAE_hyperparam_training.xlsx for some good MLP candidates.

Also happy to see you've forked the repo and started setting up a testing pipeline, once you've got something working I'd be happy for you to merge commits into main or otherwise send me the files for others to use!

Hope this helps!
Kind regards, 
Ben

**Comment by zh1jaw** [comment]
Hi Ben,

Thanks for sharing the pretrained model and the update—much appreciated! I’ll let you know if I manage to set up anything useful with it.

Best regards,
Jaw
